### PR TITLE
Updated version of java from 11 to 17

### PR DESCRIPTION
**Pull request by mskacelik**
/cc @jmartisk 

Since WF 35 will support Java 17.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [17]
+        java: [21]
```
Comment (jmartisk) [inline .github/workflows/build.yml:50]: I'd keep both 17 and 21 as a matrix to be sure, both for WildFly and Quarkus

**Comment by mskacelik** [comment]
I am going to make an issue of the weird behavior of the record containing a `List` instance after approval.

**Comment by mskacelik** [comment]
> * remove references to it in the release job (`./.github/workflows/release.yml`)

I hope that I have done it correctly.

**Review by jmartisk — APPROVED** [review]
LGTM

**Pull request merged**